rosidl_typesupport_introspection_cpp/include/rosidl_typesupport_introspection_cpp/message_introspection.hpp

```diff
@@ -27,19 +27,43 @@
 namespace rosidl_typesupport_introspection_cpp
 {
 
+/// Structure used to describe a single field of an interface type.
 typedef struct ROSIDL_TYPESUPPORT_INTROSPECTION_CPP_PUBLIC MessageMember_s
 {
+  /// The name of the field.
   const char * name_;
+  /// The type of the field as a value of the field types enum,
+  /// e.g.rosidl_typesupport_introspection_cpp::ROS_TYPE_FLOAT
   uint8_t type_id_;
+  /// If the field is a string, the upper bound on the length of the string.
   size_t string_upper_bound_;
+  /// If the type_id_ value is rosidl_typesupport_introspection_cpp::ROS_TYPE_MESSAGE
+  /// this points to an array describing the fields of the sub-interface.
   const rosidl_message_type_support_t * members_;
+  /// True if this field is an array, false if it is a unary type. An array has the same value for
+  /// type_id_.
   bool is_array_;
+  /// If is_array_ is true, this contains the number of members in the array.
   size_t array_size_;
+  /// If is_array_ is true, this specifies if the array has a maximum size. If it is true, the
+  /// value in array_size_ is the maximum size.
   bool is_upper_bound_;
+  /// The bytes into the interface's in-memory representation that this field can be found at.
   uint32_t offset_;
+  /// If the interface has a default value, this points to it.
   const void * default_value_;
+  /// If is_array_ is true, a pointer to a function that gives the size of array members.
+  /// First argument should be a pointer to the actual memory representation of the member.
   size_t (* size_function)(const void *);
+  /// If is_array_ is true, a pointer to a function that gives a const pointer to the member of the
+  /// array indicated by index.
+  /// First argument should be a pointer to the actual memory representation of the member.
+  /// Only used for nested interface types.
```

Comment:
Looks like this is used for normal types too, so I think this nested interface types sentence should be removed

```c++
  {
    "position",  // name
    ::rosidl_typesupport_introspection_cpp::ROS_TYPE_DOUBLE,  // type
    0,  // upper bound of string
    nullptr,  // members of sub message
    true,  // is array
    0,  // array size
    false,  // is upper bound
    offsetof(sensor_msgs::msg::JointState, position),  // bytes offset in struct
    nullptr,  // default value
    size_function__JointState__position,  // size() function pointer
    get_const_function__JointState__position,  // get_const(index) function pointer
    get_function__JointState__position,  // get(index) function pointer
    resize_function__JointState__position  // resize(index) function pointer
  },
```

Reply:
removed 284d4f2